src/Citeproc/Eval.hs

```diff
@@ -1186,15 +1190,17 @@ evalItem layout (position, item) = do
                                         then fmap (transform (linkTitle url)) xs
                                         -- hyperlink the entire bib item
                                         else [Linked LinkBibItem url xs] 
+            
+             let xs'' = fmap (transform fixLinkPrefixes) xs'
```

Comment:
This line is needed because an element like `<text variable="DOI" prefix="https://doi.org/">` will be realized as a `Formatted (Formatting ..) [Linked ..]` during citation processing.  Since the `prefix` is part of the formatting, this means that without this line, you'd get an ugly link like `https://doi.org/`[`12345`](https://doi.org/12345) where only the DOI itself is part of the anchor.

It makes me think that maybe we should've added links to the `Formatting` type instead, so we can control exactly when they're rendered.  That would avoid the need to traverse the tree again.

One thing to note is that with the fix applied, elements like `<text variable="DOI" prefix="DOI: ">`, will render as **[DOI: 12345](https://doi.org/12345)** instead of **DOI: [12345](https://doi.org/12345)**.

Reply:
I think the way pandoc currently works is that these prefixes are only incorporated into the link if do not end with a space. (See `fixLinks` in T.P.Citeproc and related issue jgm/pandoc#6723).  That would be the best behavior.  Can we do that?

If we put the link in Formatting, then automatically *every* prefix *and* suffix would be linked, which is not what we want, I think.

Reply:
Hmm, interesting.  

It would be easy to wrap the `Formatting` in a `Link` only when the prefix has a trailing space.  Note that this would also automatically wrap any suffix as well (as does the current code).

The [`fixLinks`](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Citeproc.hs#L424) function actually does something more clever, though, right?  It only acts when the prefix + anchor equals the link target.  That seems a bit more difficult to reproduce here.

Reply:
Oh right, I forgot that `fixLinks` does that.  I think we should reproduce that behavior.
Your function seems good to me.

Reply:
Ok!  Just pushed the fix.